### PR TITLE
[11.x] Fix: Removed TTY mode to resolve Windows compatibility issue 

### PR DESCRIPTION
**Pull request by yourchocomate**
Error: RuntimeException - TTY mode is not supported on Windows platform.

Screenshot 2024-03-13 063237

### Description
This PR addresses a compatibility issue on Windows platforms where TTY mode is not supported. The issue arises in the installNodeDependencies() function, where TTY mode is set unnecessarily. This setting triggers a Symfony\Component\Process\Exception\RuntimeException due to unsupported TTY mode on Windows.

### Changes Made

- Removed unnecessary TTY mode from Symfony Process configuration.
- Conditionally set the TTY mode based on the operating system, ensuring compatibility with both Windows and non-Windows platforms.
- Improved error handling and user feedback in case of failed command execution.

### CODE REVIEW

File: src/Illuminate/Foundation/Console/BroadcastingInstallCommand.php [hunk 1]
```diff
@@ -168,9 +168,17 @@ protected function installNodeDependencies()
...
         }
 
-        Process::command(implode(' && ', $commands))
-            ->path(base_path())
-            ->tty(true)
-            ->run();
+        $command = Process::command(implode(' && ', $commands))
+                        ->path(base_path());
+
+        if (strtoupper(substr(PHP_OS, 0, 3)) !== 'WIN') {
```
Comment (crynobone) [inline src/Illuminate/Foundation/Console/BroadcastingInstallCommand.php:174]:
```suggestion
        if (! windows_os()) {
```

`windows_os` function available from `Illuminate\Support\helpers.php`.

File: src/Illuminate/Foundation/Console/BroadcastingInstallCommand.php [hunk 2]
```diff
@@ -168,9 +168,17 @@ protected function installNodeDependencies()
...
-            ->tty(true)
-            ->run();
+        $command = Process::command(implode(' && ', $commands))
+                        ->path(base_path());
+
+        if (! windows_os()) {
+            $command->tty(true);
+        }
+
+        if($command->run()->failed()) {
```
Comment (driesvints) [inline src/Illuminate/Foundation/Console/BroadcastingInstallCommand.php:178]:
```suggestion
        if ($command->run()->failed()) {
```

File: src/Illuminate/Foundation/Console/stubs/echo-bootstrap-js.stub [hunk 3]
```diff
@@ -4,4 +4,4 @@
  * allow your team to quickly build robust real-time web applications.
  */
 
-import './echo';
+import './echo.js';
```
Comment (menglisch) [inline src/Illuminate/Foundation/Console/stubs/echo-bootstrap-js.stub:7]: What does this line change fix?

Comment (driesvints) [inline src/Illuminate/Foundation/Console/stubs/echo-bootstrap-js.stub:7]: Removing this for now. @yourchocomate please send in a separate PR with explanation if you need this.

```suggestion
```

**Review by driesvints — APPROVED** [review]

File: src/Illuminate/Foundation/Console/stubs/echo-bootstrap-js.stub [hunk 4]
```diff
@@ -4,4 +4,3 @@
  * allow your team to quickly build robust real-time web applications.
  */
```
Comment (driesvints) [inline src/Illuminate/Foundation/Console/stubs/echo-bootstrap-js.stub:6]:
```suggestion

import './echo';
```

File: src/Illuminate/Foundation/Console/stubs/echo-bootstrap-js.stub (hunk 3 above)
Comment (yourchocomate) [reply src/Illuminate/Foundation/Console/stubs/echo-bootstrap-js.stub]: Mistakenly pushed to this PR. Since it's related to broadcast installation, I kept the commit. The bug is explained below:

**Problem**: The installation keeps inserting import of echo in bootrap.js

**Buggy Code:**
`if (! str_contains($bootstrapScript, 'echo.js'))`

The import checker checks for echo.js, that's okay but the stub not follows the checker strategy, which makes the checker always return false

Thus, repeating import of echo.js for every `install:broadcasting` trial

Comment (yourchocomate) [reply src/Illuminate/Foundation/Console/stubs/echo-bootstrap-js.stub]: Created a PR  #50520 regarding the issue I mentioned , you can check this out. Thanks
